src/jdk.compiler/share/classes/com/sun/tools/javac/main/JavaCompiler.java

```diff
@@ -1567,6 +1568,12 @@ public void visitSwitchExpression(JCSwitchExpression tree) {
                 hasPatterns |= tree.patternSwitch;
                 super.visitSwitchExpression(tree);
             }
+
+            @Override
+            public void visitDerivedInstance(JCDerivedInstance tree) {
+                hasPatterns |= true;
```

Comment:
it could be thought as a pattern but it is not a pattern nor it is internally converted to a pattern, unless I'm missing something